Makefile

```diff
@@ -1,13 +1,73 @@
-# Warning
+# This Makefile is (and must be) used by
+# travis/pre-commit.sh to qualify pull requests.
 #
-# This makefile is provided only as a convenience to run
-# the travis pre-commit script, which generates all the
-# code that needs to be generated, and runs all the tests.
+# That script generates all the code that needs
+# to be generated, and runs all the tests.
 #
-# This makefile won't be maintained until it replaces
-# (or is used by) the travis/pre-commit.sh script
+# Functionality in that script, expressed in bash, is
+# gradually being moved here.
 
+MYGOBIN := $(shell go env GOPATH)/bin
+
+.PHONY: all
 all:
 	./travis/pre-commit.sh
 
-.PHONY: all
+$(MYGOBIN)/golangci-lint:
+	cd api; go install github.com/golangci/golangci-lint/cmd/golangci-lint
```

Comment:
Q: why `cd api` ? 
May be using `go get` in a temporary directory to install a semver'd golangci-lint like we are doing with other tools like helm/kubeval ?

I think `go install` will probably install it from HEAD, we probably want to avoid doing that to prevent travis surprises.

Reply:
The `cd api` is to allow the versions specified in `api/go.mod` the to be respected.

`go install` won't install from head, it will install the version specified in the `go.mod`.
I just don't want a go.mod at the top level, as it would be misleading.

Which begs the question: why not install helm and kubeval this way?  Because - having tried it - i don't trust their go.mod adoption/setup for binary-only installation yet.  If that improves, we can switch.  Further, helm and kubeval are only used in two example plugins, not the core functionality.  I'd delete the examples entirely at this stage if it weren't possible to download binaries and use them in at least one test scenario.

Reply:
BTW, this is a great thing about go.mod and programs that respect it.
kustomize uses v1.19.1 of golanci-lint because 1.20 had some nasty (to me) defaults

Reply:
aah...ok. got it. So `go install` respects go.mod of the `api` module. That's why we added `tools.go` with the naked imports. 

This all makes sense.